crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -542,12 +544,60 @@ async fn execute_transactions(
     authority_state
         .database
         .epoch_store()
-        .insert_pending_certificates(&txns)?;
-
-    authority_state.transaction_manager.enqueue(txns).await?;
+        .insert_pending_certificates(&synced_txns)?;
 
     authority_state
+        .transaction_manager
+        .enqueue(synced_txns)
+        .await?;
+
+    let effects = authority_state
+        .database
+        .notify_read_effects(synced_tx_digests)
+        .await?;
+
+    assert_eq!(
+        effects.len(),
+        num_synced_txns,
```

Comment:
i think this should `execution_digests.len()` - I believe they will be the same in this version of the code, but it takes some work to convince oneself of that, and I think it would be better to make the assert more future-proof against any changes to the code in the future.

Reply:
I actually made a more foundational change to how I'm checking this invariant based on previous comments. Rather than check for length equality, I just await for effects for all input digests. This is the ultimate test as you mentioned.